kansha/column/comp.py

```diff
@@ -56,13 +56,17 @@ def __init__(self, id_, board, card_extensions, search_engine, services_service,
 
     def copy(self, parent, additional_data):
         new_data = self.data.copy(parent.data)
-        new_obj = self._services(Column, new_data.id, None, self.card_extensions, self.search_engine, data=new_data)
-
+        new_column = self._services(Column, new_data.id, None, self.card_extensions, self.search_engine, data=new_data)
         for card in self.cards:
-            new_card = card().copy(new_obj, additional_data)
-            new_obj.cards.append(component.Component(new_card))
-
-        return new_obj
+            new_card = card().copy(new_column, additional_data)
+            new_column.index_card(new_card)
```

Comment:
**Beware!** you are committing at each iteration. That's very inefficient. You should commit once, after adding all documents to index.